pkg/model/hornet/message_id.go

```diff
@@ -0,0 +1,148 @@
+package hornet
+
+import (
+	"encoding/hex"
+	"fmt"
+	"sort"
+
+	iotago "github.com/iotaledger/iota.go/v2"
+)
+
+// MessageID is the ID of a Message.
+type MessageID []byte
+
+// MessageIDs is a slice of MessageID.
+type MessageIDs []MessageID
+
+// ToHex converts the MessageID to its hex representation.
+func (m MessageID) ToHex() string {
+	return hex.EncodeToString(m)
+}
+
+// ToArray converts the MessageID to an array.
+func (m MessageID) ToArray() iotago.MessageID {
+	var messageID iotago.MessageID
+	copy(messageID[:], m)
+	return messageID
+}
+
+// ToMapKey converts the MessageID to a string that can be used as a map key.
+func (m MessageID) ToMapKey() string {
+	return string(m)
+}
+
+// GetNullMessageID returns the ID of the genesis message.
+func GetNullMessageID() MessageID {
+	nullMessageID := make(MessageID, 32)
+	return nullMessageID
+}
+
+// MessageIDFromHex creates a MessageID from a hex string representation.
+func MessageIDFromHex(hexString string) (MessageID, error) {
+
+	b, err := hex.DecodeString(hexString)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(b) != iotago.MessageIDLength {
+		return nil, fmt.Errorf("unknown messageID length (%d)", len(b))
+	}
+
+	return MessageID(b), nil
+}
+
+// MessageIDFromMapKey creates a MessageID from a map key representation.
+func MessageIDFromMapKey(mapKey string) MessageID {
+	if len(mapKey) != iotago.MessageIDLength {
+		panic(fmt.Sprintf("unknown messageID length (%d)", len(mapKey)))
+	}
+
+	return MessageID(mapKey)
+}
+
+// MessageIDFromSlice creates a MessageID from a byte slice.
+func MessageIDFromSlice(b []byte) MessageID {
+
+	if len(b) != iotago.MessageIDLength {
+		panic(fmt.Sprintf("unknown messageID length (%d)", len(b)))
+	}
+
+	return MessageID(b)
+}
+
+// MessageIDFromArray creates a MessageID from a byte array.
+func MessageIDFromArray(b iotago.MessageID) MessageID {
+	return MessageID(b[:])
```

Comment:
```suggestion
	return append(MessageID{}, b[:]...)
```
I would seriously consider copying the actual iotago.MessageID and not just pointing to it.
This has a minimal performance penalty when b is indeed not mutated, but since it can cause nasty bugs (like the loop variable) it might be worth it.